pkg/xsd/workspace.go

```diff
@@ -40,9 +40,33 @@ func (ws *Workspace) loadXsd(xsdPath string) (*Schema, error) {
 	}
 	schema.ModulesPath = ws.GoModulesPath
 	schema.filePath = xsdPath
-	ws.Cache[xsdPath] = schema
+	// Won't cache included schemas - we need to append contents to the current
+	// schema.
+	if cache {
+		ws.Cache[xsdPath] = schema
+	}
 
 	dir := filepath.Dir(xsdPath)
+
+	for idx, _ := range schema.Includes {
+		si := schema.Includes[idx]
+		if err := si.load(ws, dir); err != nil {
+			return nil, err
+		}
+
+		isch := si.IncludedSchema
+		schema.Imports = append(isch.Imports, schema.Imports...)
+		schema.Elements = append(isch.Elements, schema.Elements...)
+		schema.Attributes = append(isch.Attributes, schema.Attributes...)
+		schema.AttributeGroups = append(isch.AttributeGroups, schema.AttributeGroups...)
+		schema.ComplexTypes = append(isch.ComplexTypes, schema.ComplexTypes...)
+		schema.SimpleTypes = append(isch.SimpleTypes, schema.SimpleTypes...)
+		schema.inlinedElements = append(isch.inlinedElements, schema.inlinedElements...)
```

Comment:
Looking at these list appends.

Do we want to de-duplicate the elements? I mean may it happen that the current schema re-defines field of the  included schema?

Reply:
I believe that should not happen - XSD `include` definitions should work in the same way as if the definition is copy-pasted from the included XSD to the XSD which included it. The same situation should occur if we define the same element or type twice in a single XSD file.